settings/defaults.php

```diff
@@ -1,6 +1,6 @@
 <?php
-@define('CONST_BasePath', '@CMAKE_SOURCE_DIR@');
-@define('CONST_InstallPath', '@CMAKE_BINARY_DIR@');
+@define('CONST_BasePath', '@WWW_APP@');
+@define('CONST_InstallPath', '@WWW_PUBLIC@');
```

Comment:
No worries about the negative, I can take it. Sometimes the best way to get something done is to start doing it ;).

I looked at the difference between install and base and figured that Install Path is in the webroot and base path are included files (classes and so on). I may have that backwards, but maybe renaming the constants to reflect what they are semantically (not where they reside) is a good idea to begin with.
Supporting both installed and "staged" would be hard to do if you want to keep it in one file. It's probably better create two versions  of settings.php. One that works in-place as is and one that is going to be installed.